### DISCUSSION TITLE
System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'

### DISCUSSION THREAD

**Discussion by LiorBanaiKama** [pythonnet/pythonnet#1821] [General]
Hi,
What could cause this exception when calling pythonnet  after 4 times?

![image](https://user-images.githubusercontent.com/73282711/173868310-b7c47a5b-9f3b-44cf-8fa4-e7b84782a79f.png)

I'll add a small test project soon

**Comment by lostmsu** [comment]
`AccessViolationException` is generally a sign that you are not holding GIL.

**Reply by LiorBanaiKama** [reply]
@lostmsu thanks.
Apparently the issue is caused by calling plot method from C#:

import matplotlib.pyplot as plt
..
def plot(af: AF):
    print('plot called')
    af.plot("AF")
    plt.pause(1)
    plt.show()

I guess I need to release the plot resources somehow (the method is blocked until I close the plot).
